rct_optimizations/include/rct_optimizations/validation/homography_validation.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include <rct_optimizations/types.h>
+
+namespace rct_optimizations
+{
+
+/**
+ * @brief A simple structure for choosing which correspondence indices to use for generating the homography matrix
+ */
+struct CorrespondenceSampler
+{
+  virtual std::vector<std::size_t> getSampleCorrespondenceIndices() const = 0;
+};
+
+/**
+ * @brief A correspondence sampler specifically for modified circle grid targets
+ */
+struct ModifiedCircleGridCorrespondenceSampler : CorrespondenceSampler
+{
+  ModifiedCircleGridCorrespondenceSampler(const std::size_t rows_, const std::size_t cols_)
+    : rows(rows_)
+    , cols(cols_)
+  {
+  }
+
+  virtual std::vector<std::size_t> getSampleCorrespondenceIndices() const final override
+  {
+    // Make sure there are at least two times as many points as the number of sample points
+    if ((rows * cols / 2) < 8)
```

Comment:
Do we still want to have a variable number of points used, or are we comfortable fixing it to 8?

Reply:
I just pushed a change to use the minimum number of points (4, instead of 8). This struct is easy enough to override, so I don't think we necessarily need to make it a parameterizable number of points